xbmc/pvr/epg/Epg.cpp

```diff
@@ -145,8 +145,8 @@ void CPVREpg::Clear(void)
 
 void CPVREpg::Cleanup(void)
 {
-  CDateTime cleanupTime = CDateTime::GetCurrentDateTime().GetAsUTCDateTime() -
-      CDateTimeSpan(0, g_advancedSettings.m_iEpgLingerTime / 60, g_advancedSettings.m_iEpgLingerTime % 60, 0);
+  int iPastDays = CServiceBroker::GetSettings().GetInt(CSettings::SETTING_EPG_PAST_DAYSTODISPLAY);
+  const CDateTime cleanupTime = CDateTime::GetUTCDateTime() - CDateTimeSpan(iPastDays, 0, 0, 0);
```

Comment:
could we unify initialization and use `cleanupTime(value)` like you do all over the place

Reply:
For modern C++ it should be quite the other way round. "default construction + assignment" is more readable code and with modern compilers there is no need any more to use "copy construction". Formerly, I did this for performance reasons. Modern compilers will elide this - no need for such tricks anymore. So, my code evolves over time.  Changing all occurrences of this pattern definitely is for another day.

Reply:
Sure